common/unified/matrix/dense_kernels.cpp

```diff
@@ -180,6 +180,39 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_SCALAR_TYPE(
     GKO_DECLARE_DENSE_ADD_SCALED_KERNEL);
 
 
+template <typename ValueType, typename ScalarType>
+void add_scale(std::shared_ptr<const DefaultExecutor> exec,
+               const matrix::Dense<ScalarType>* alpha,
+               const matrix::Dense<ValueType>* x,
+               const matrix::Dense<ScalarType>* beta,
+               matrix::Dense<ValueType>* y)
+{
```

Comment:
Also maybe make sure that y and x both have non-zero sizes ? If input has zero size, then you can fill the output with zero ?

Reply:
I guess it should be fine because in core, we assert that x and y should have the same size. If the output has some finite size and the input is 0x0, that would be an error.

Reply:
Yes, with the changes/semantics that Tobias implemented, using a 0x0 matrix is not necessarily an error. For example [see dense apply](https://github.com/ginkgo-project/ginkgo/blob/develop/cuda/matrix/dense_kernels.cu#L164=)